### ISSUE TITLE
Exception when loading 5.18.0

### ISSUE THREAD

**Issue by TimvdLippe** [jshiell/checkstyle-idea#385]
When I install CheckStyle-IDEA 5.18.0, I receive a `java.lang.NoSuchMethodException` in the `CheckFactory`:

```
2018-03-07 14:14:10,022 [  17854]   WARN - checkstyle.checks.CheckFactory - Couldn't instantiate check class class org.infernus.idea.checkstyle.checks.JavadocPackageCheck 
java.lang.NoSuchMethodException: org.infernus.idea.checkstyle.checks.JavadocPackageCheck.<init>(com.intellij.openapi.project.Project)
	at java.lang.Class.getConstructor0(Class.java:3082)
	at java.lang.Class.getConstructor(Class.java:1825)
	at org.infernus.idea.checkstyle.checks.CheckFactory.getChecks(CheckFactory.java:31)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.createListener(OpScan.java:86)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:52)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:22)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:130)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.scan(CheckstyleActionsImpl.java:76)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.scan(CheckStyleChecker.java:45)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$inspectFile$1(CheckStyleInspection.java:91)
	at java.util.Optional.map(Optional.java:215)
	at org.infernus.idea.checkstyle.CheckStyleInspection.inspectFile(CheckStyleInspection.java:91)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$checkFile$0(CheckStyleInspection.java:63)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.call(ApplicationImpl.java:338)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

It appears that these changes (https://github.com/jshiell/checkstyle-idea/commit/ff93cd57797bf5a7260596a038f10f38f46bca26#diff-1b92cf48d542ee63fdbb37e2c4573743R27) are breaking https://github.com/jshiell/checkstyle-idea/blob/647e3e83ac820376e729b5b35920e874486e7fd6/src/main/java/org/infernus/idea/checkstyle/checks/CheckFactory.java#L31 The custom class loading in `CheckFactory` is looking for a constructor with type `Project`, but instead it finds the `CheckstyleProjectService` there.

My IntelliJ now complains about corrupt classpaths, which only started complaining after installing the latest version. Reverting to 5.17.0 seems to fix the issue.

**Comment by jshiell** [comment]
Thanks for the report - I'll try and have a look at this this weekend.

**Comment by jshiell** [comment]
Sorry about that. Should be fixed in 5.18.2.

**Comment by TimvdLippe** [comment]
@jshiell Awesome, thanks for the quick turnaround!
